### PR TITLE
Add MiskTracer to facilitate ad-hoc method tracing

### PR DESCRIPTION
**Pull request by nbroyles**

### CODE REVIEW

File: misk/src/main/kotlin/misk/tracing/MiskTracer.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  private val tracer: Tracer
+) {
+  /**
+   * Adds span/sub-span to trace for method provided as a parameter. Use this method to do ad-hoc
+   * tracing on methods you're interested in.
+   *
+   * @param operationName name to be used for the span. Usually, should be a string representation
+   * of the method you plan to trace
+   */
+  fun trace(operationName: String, method: () -> Any?) {
```
Comment (mmihic) [inline misk/src/main/kotlin/misk/tracing/MiskTracer.kt]: This should return the results from the method, and the signature should have a generic type bounds. e.g.

````kotlin
fun <R> trace(operationName: String, method: () -> R) : R = 
   tracer.buildSpan(operationName).startActive(true).use { method() }
````

File: misk/src/main/kotlin/misk/tracing/MiskTracer.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import javax.inject.Inject
+
+class MiskTracer @Inject internal constructor(
+  private val tracer: Tracer
+) {
+  /**
+   * Adds span/sub-span to trace for method provided as a parameter. Use this method to do ad-hoc
+   * tracing on methods you're interested in.
+   *
+   * @param operationName name to be used for the span. Usually, should be a string representation
```
Comment (mmihic) [inline misk/src/main/kotlin/misk/tracing/MiskTracer.kt]: nit: call the parameter ``spanName`` to better reflect its meaning

File: misk/src/main/kotlin/misk/tracing/MiskTracer.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package misk.tracing
+
+import io.opentracing.Tracer
+import javax.inject.Inject
+
+class MiskTracer @Inject internal constructor(
```
Comment (mmihic) [inline misk/src/main/kotlin/misk/tracing/MiskTracer.kt]: nit: Don't preface classes with ``Misk`` unless absolutely necessary.

File: misk/src/main/kotlin/misk/tracing/MiskTracer.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+) {
+  /**
+   * Adds span/sub-span to trace for method provided as a parameter. Use this method to do ad-hoc
+   * tracing on methods you're interested in.
+   *
+   * @param operationName name to be used for the span. Usually, should be a string representation
+   * of the method you plan to trace
+   */
+  fun trace(operationName: String, method: () -> Any?) {
+    tracer.buildSpan(operationName).startActive(true).use { method() }
```
Comment (mmihic) [inline misk/src/main/kotlin/misk/tracing/MiskTracer.kt]: Catch exceptions and set ERROR tag?

File: misk/src/main/kotlin/misk/tracing/MiskTracer.kt (hunk 3 above)
Comment (nbroyles) [reply misk/src/main/kotlin/misk/tracing/MiskTracer.kt]: Only doing it here because there are so many other overloaded uses of simply `Tracer`

**Pull request merged**